staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -534,14 +534,14 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]*w
 	size := w.endIndex - w.startIndex
 	var oldest uint64
 	switch {
+	case w.listResourceVersion > 0 && w.startIndex == 0:
```

Comment:
Is `w.startIndex == 0` used to judge whether watch cache expand normally?
Will any issue happen if watch cache becomes full again after a shrink?(Seems this should be covered in TestCacheIncreaseDoesNotBreakWatch)

Reply:
How about `w.listResourceVersion > 0 && w.capacity == w.upperBoundCapacity`